node/metrics/createQuote.ts

```diff
@@ -0,0 +1,83 @@
+import type { Metric } from './metrics'
+import { sendMetric } from './metrics'
+
+type UserData = {
+  orgId: string
+  costId: string
+  roleId: string
+}
+
+type SessionData = {
+  namespaces: {
+    profile: {
+      id: { value: string }
+      email: { value: string }
+    }
+    account: {
+      accountName: { value: string }
+    }
+  }
+}
+
+type CreateQuoteMetricParam = {
+  sessionData: SessionData
+  sendToSalesRep: boolean
+  userData: UserData
+  quoteId: string
+  quoteReferenceName: string
+  creationDate: string
+}
+
+type CreateQuoteFieldsMetric = {
+  cost_center_id: string
+  buyer_org_id: string
+  member_id: string
+  member_email: string
+  role: string
+  creation_date: string
+  quote_id: string
+  quote_reference_name: string
+  send_to_sales_rep: boolean
+}
+
+type CreateQuoteMetric = Metric & { fields: CreateQuoteFieldsMetric }
+
+const buildQuoteMetric = (
+  metricsParam: CreateQuoteMetricParam
+): CreateQuoteMetric => {
+  const { namespaces } = metricsParam.sessionData
+  const accountName = namespaces.account.accountName.value
+  const userEmail = namespaces.profile.email.value
```

Comment:
Similar thing here: you're assuming on line 60 further below that `namespaces` and `profile` can be null/undefined. Better to assume the same here.